test/test_fs.py

```diff
@@ -268,6 +271,7 @@ def run_fs(mountpoint, cross_process):
     testfs = Fs(cross_process)
     fuse_options = set(pyfuse3.default_options)
     fuse_options.add('fsname=pyfuse3_testfs')
+    fuse_options.add('max_read=4096')
```

Comment:
Could you also test if it has the desired effect, i.e. lengths are actually limited to 4k?

Reply:
When `max_read=N` is used without `conn_info.max_read = N` (or vice versa) fuse terminates and many tests fail. So, assuming that previous code works (setting fuse options), there is some assurance that fuse is receiving our conn_info.max_read value.

That said I may write a test that reads are really limited, but this will delay this PR.